gammapy/modeling/models/core.py

```diff
@@ -361,6 +362,46 @@ def to_dict(self):
         else:
             return {"components": models_data}
 
+    def to_parameters_table(self):
+        """Convert Models parameters to an astropy Table."""
+        pars = self.parameters.to_table()
+        rows = []
+        for name in self.parameters_unique_names:
+            row = {}
+            #Warning: splitting of parameters will break is source name has a "." in its name.
+            parts = name.split(".")
+            if len(parts) == 3:
+                row["model"], row["type"], _ = parts
+            elif len(parts) == 2:
+                row["model"], row["type"] = parts[0], ""
+            rows.append(row)
+        table = table_from_row_data(rows)
+        table = hstack([table, pars])
+        self._table_cached = table
+        return  table
+
+    def update_from_table(self, t):
+        """Update Models from an astropy Table."""
+        self._check_non_editable_columns(t, self._table_cached)
+        parameters_dict = [dict(zip( t.colnames, row)) for row in t]
+        for k, data in enumerate(parameters_dict):
+            self._update_from_dict(self.parameters[k], data)
+
+    def _update_from_dict(self, parameter, data):
```

Comment:
As this is a private function anyway you could directly rely on `Parameter.__dict__.update()` and remove this function...

Reply:
I'm back having a look at this unfinished PR. I tried to update the parameters using  `__dict__.update` but it doesn't seem to update the parameters in the end. 
```
    for k, data in enumerate(parameters_dict):
        if k ==0 : print(self.parameters[k], data)
#        self._update_from_dict(self.parameters[k], data) # previous way
        self.parameters[k].__dict__.update(data) # new. is this ok? doesn't crash bu doesn't update the values
```

Reply:
Fixed adding `update_from_dict` to parameters in PR #3137